--- a/src/apps_mem_imp.c
+++ b/src/apps_mem_imp.c
@@ -101,6 +101,7 @@
 __QAIC_IMPL(apps_mem_request_map64)(int heapid, uint32_t lflags, uint32_t rflags,
                                     uint64_t vin, int64_t len, uint64_t *vapps,
                                     uint64_t *vadsp) __QAIC_IMPL_ATTRIBUTE {
+  if (rflags == 0) rflags = 0x1000;
   struct mem_info *minfo = 0;
   int nErr = 0, unsigned_module = 0, ualloc_support = 0;
   void *buf = 0;
